plugins:foreman_azure - Release 1.0.1 (DEB)

dLobatog opened the pull request.

domcleal commented on plugins/ruby-foreman-azure/debian/control:
```diff
@@ -0,0 +1,13 @@
+
+Package: ruby-foreman-azure
+Architecture: all
+Depends: ${misc:Depends}, bundler, foreman (>= 1.11.0), ruby-foreman-deface (<<2.0.0), foreman-compute(>= 1.11.0)
```
foreman-compute shouldn't be required if the plugin only uses fog-azure.  (foreman-compute ships all of Fog)

domcleal replied:
also the gem states >= 1.8, not >= 1.11

domcleal commented on plugins/ruby-foreman-azure/debian/gem.list:
```diff
@@ -0,0 +1,2 @@
+# list of gem urls to download via Jenkins, space separated
+GEMS="https://rubygems.org/downloads/foreman_azure-1.0.0.gem"
```
This will also need every dependency of foreman_azure - i.e. fog-azure, azure (and recurse!)

dLobatog replied:
Fixed, I think

domcleal commented on plugins/ruby-foreman-azure/debian/copyright:
```diff
@@ -0,0 +1,7 @@
+Source: https://github.com/theforeman/foreman_azure
+
+Files: *
+Copyright: 2015 Daniel Lobato Garcia <elobatocs@gmail.com>
```
2016

domcleal commented on plugins/ruby-foreman-azure/debian/rules:
```diff
@@ -0,0 +1,13 @@
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+%:
```
Seems to be missing asset compilation, the gem appears to have a JS asset that'll need precompiling.

domcleal commented on plugins/ruby-foreman-azure/debian/gem.list:
```diff
@@ -0,0 +1,13 @@
+GEMS="$GEMS https://rubygems.org/downloads/azure-0.7.5.gem"
+GEMS="$GEMS https://rubygems.org/downloads/azure-core-0.1.1.gem"
+GEMS="$GEMS https://rubygems.org/downloads/faraday_middleware-0.10.0.gem"
+GEMS="$GEMS https://rubygems.org/downloads/json-1.8.3.gem"
```
already in Foreman

domcleal commented on plugins/ruby-foreman-azure/debian/gem.list:
```diff
@@ -0,0 +1,13 @@
+GEMS="$GEMS https://rubygems.org/downloads/azure-core-0.1.1.gem"
+GEMS="$GEMS https://rubygems.org/downloads/faraday_middleware-0.10.0.gem"
+GEMS="$GEMS https://rubygems.org/downloads/json-1.8.3.gem"
+GEMS="$GEMS https://rubygems.org/downloads/mime-types-3.0.gem"
```
ditto

domcleal commented on plugins/ruby-foreman-azure/debian/gem.list:
```diff
@@ -0,0 +1,13 @@
+GEMS="$GEMS https://rubygems.org/downloads/faraday_middleware-0.10.0.gem"
+GEMS="$GEMS https://rubygems.org/downloads/json-1.8.3.gem"
+GEMS="$GEMS https://rubygems.org/downloads/mime-types-3.0.gem"
+GEMS="$GEMS https://rubygems.org/downloads/mime-types-data-3.2016.0221.gem"
```
not needed if using mime-types 1.25.1 from Foreman

domcleal commented on plugins/ruby-foreman-azure/debian/gem.list:
```diff
@@ -0,0 +1,13 @@
+GEMS="$GEMS https://rubygems.org/downloads/mime-types-3.0.gem"
+GEMS="$GEMS https://rubygems.org/downloads/mime-types-data-3.2016.0221.gem"
+GEMS="$GEMS https://rubygems.org/downloads/systemu-2.6.5.gem"
+GEMS="$GEMS https://rubygems.org/downloads/thor-0.19.1.gem"
```
in Foreman too

domcleal commented on plugins/ruby-foreman-azure/debian/control:
```diff
@@ -0,0 +1,13 @@
+Section: ruby
+Priority: extra
+Maintainer: Daniel Lobato Garcia <elobatocs@gmail.com>
+Build-Depends: debhelper, foreman (>= 1.11.0), ruby-foreman-deface (<<2.0.0)
```
missing foreman-assets, foreman-sqlite3 for build deps, in order to precompile assets (cause of the test failure)

dLobatog commented:
Updated to remove all the gems provided by the SCL (mime-types, json) and thor. I also added foreman-assets and sqlite3 as a dependency to precompile assets properly.

Where's thor coming from? rh-ruby22?

domcleal commented:
> Where's thor coming from? rh-ruby22?

Unlikely in a Debian build!  It's a dependency of railties.

domcleal commented on plugins/ruby-foreman-azure/debian/control:
```diff
@@ -0,0 +1,13 @@
+
+Package: ruby-foreman-azure
+Architecture: all
+Depends: ${misc:Depends}, bundler, foreman (>= 1.11.0), foreman-assets (>= 1.11.0), foreman-sqlite3 (>= 1.11.0), ruby-foreman-deface (<<2.0.0)
```
assets and sqlite3 should only be added to build-deps for asset precompilation, they're not runtime deps

domcleal commented on plugins/ruby-foreman-azure/debian/install:
```diff
@@ -0,0 +1,2 @@
+foreman_azure.rb  usr/share/foreman/bundler.d
+cache                usr/share/foreman/vendor
```
also needed here for assets:

foreman_azure                 var/lib/foreman/public/assets

domcleal commented on plugins/ruby-foreman-azure/debian/gem.list:
```diff
@@ -0,0 +1,9 @@
+GEMS="$GEMS https://rubygems.org/downloads/faraday_middleware-0.10.0.gem"
+GEMS="$GEMS https://rubygems.org/downloads/systemu-2.6.5.gem"
+GEMS="$GEMS https://rubygems.org/downloads/uuid-2.3.8.gem"
+gems="$GEMS https://rubygems.org/downloads/macaddr-1.7.1.gem"
```
GEMS should be caps, I think this is causing the build failure:

<pre>
Resolving dependencies...
Some gems seem to be missing from your vendor/cache directory.
Could not find gem 'macaddr (~> 1.0) ruby', which is required by gem
'foreman_azure (= 1.0.0) ruby', in any of the sources.
</pre>

dLobatog commented:
There's a typo on the engine, so I guess it'll have to be 1.0.1 - NameError: uninitialized constant ForemanAzure::Engine::SETTING

dLobatog commented:
Success, finally :smile:  - it's 1.0.1

domcleal commented:
Works well, thanks! Just to check, would you like this in 1.11, 1.12 and nightly?

domcleal commented:
Merged, thanks @dLobatog.  Building for 1.11 and nightly now, 1.12 on RC1's release.
